### PR TITLE
Add submodule init and update

### PR DESCRIPTION
**Pull request by yogeshlonkar**
Currently submodules in git repositories are not initialized or updated
when cloned, this update will perform submodule init and update

https://travix.atlassian.net/browse/TL-28

### CODE REVIEW

File: gitRunner.go [hunk 1]
```diff
@@ -88,6 +90,26 @@ func gitClone(ctx context.Context, gitName, gitURL, gitBranch string, shallowClo
 
 	err = foundation.RunCommandWithArgsExtended(ctx, "git", args)
 
+	if err != nil {
+		return
+	}
+
+	// submodule init and update
+	cmd := exec.CommandContext(ctx, "git", "submodule", "init")
```
Comment (msharbaji) [inline gitRunner.go]: @yogeshlonkar I'd say it's better to use estafette-foundation to execute the command, so it's better to have the consistence repository

**Comment by yogeshlonkar** [comment]
Tested in https://estafette.travix.com/pipelines/bitbucket.org/xivart/spring-boot-demo-ylonkar/builds/752896401425596436/logs/752896673348157461

**Review by msharbaji — APPROVED** [review]

File: gitRunner.go (hunk 1 above)
Comment (yogeshlonkar) [reply gitRunner.go]: I think its from older code as the commands were replaced with foundation

File: gitRunner.go [hunk 2]
```diff
@@ -91,6 +94,53 @@ func gitClone(ctx context.Context, gitName, gitURL, gitBranch string, shallowClo
...
+		}
+
+		log.Info().Msg("Updating submodules")
+		err = foundation.RunCommandInDirectoryWithArgsExtended(ctx, targetDirectory, "git", []string{"submodule", "update"})
+		if err != nil {
+			return
+		}
+
+		// restore .gitmodules file
+		err = foundation.RunCommandInDirectoryWithArgsExtended(ctx, targetDirectory, "git", []string{"checkout", ".gitmodules"})
```
Comment (markvincze) [inline gitRunner.go:139]: @yogeshlonkar just out of curiosity: why is this extra checkout command needed? Isn't this file checked out already when the original `git clone` is being done?

Comment (yogeshlonkar) [reply gitRunner.go:139]: @markvincze well for `git submodule update` to run successfully I had to modify .gitmodules by adding tokens to URLs for submodules. Once submodule is updated I'm reverting the changes to .gitmodules

Comment (markvincze) [reply gitRunner.go:139]: Oh, right, I forgot about that, thanks for the explanation!

**Pull request merged**